echopype/convert/api.py

```diff
@@ -433,9 +433,21 @@ def open_raw(
 
         # TODO: perform more robust tests for the 'auto' heuristic value
         if offload_to_zarr == "auto" and p2z.write_to_zarr(mem_mult=0.4):
+
             p2z.datagram_to_zarr(max_mb=max_zarr_mb)
+
+            # rectangularize complex transmit ping data, if it exists
+            if "complex" in parser.ping_data_dict_tx.keys():
+                parser.rectangularize_transmit_ping_data(data_type="complex")
```

Comment:
Looking at the parser code here:
https://github.com/OSOceanAcoustics/echopype/blob/0c117927a9ef17055f1ebe2b84af1eebad1e55f4/echopype/convert/utils/ek_raw_parsers.py#L1671-L1674
It seems that "complex" would always exist, just whether or not it holds actual value.

And in fact under your current `self.rectangularize_transmit_ping_data(data_type)` this is not checked and everything still run through.

Reply:
As we discussed, we will not change this for now.